pyro/distributions/torch_patch.py

```diff
@@ -80,6 +80,16 @@ def _HalfCauchy_logprob(self, value):
     return log_prob
 
 
+# TODO fix batch_shape have an extra singleton dimension upstream
```

Comment:
this [upstream line](https://github.com/pytorch/pytorch/blob/master/torch/distributions/constraints.py#L467) is not correct IMO.

Reply:
yeah weird, I don't understand the `.unsqueeze(0)`.